common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt

```diff
@@ -0,0 +1,227 @@
+
+    private val binding: ViewNotifyMeViewBinding by viewBinding()
+
+    private val viewModel by lazy {
+        val factory = NotifyMeViewModel.Factory(findViewTreeSavedStateRegistryOwner()!!)
+        ViewModelProvider(findViewTreeViewModelStoreOwner()!!, factory)[NotifyMeViewModel::class.java]
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        ViewTreeLifecycleOwner.get(this)?.lifecycle?.addObserver(viewModel)
+
+        @SuppressLint("NoHardcodedCoroutineDispatcher")
+        coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
+
+        viewModel.setNotifyMeListener(listener)
+
+        viewModel.viewState
+            .onEach { render(it) }
+            .launchIn(coroutineScope!!)
+
+        viewModel.commands()
+            .onEach { processCommands(it) }
+            .launchIn(coroutineScope!!)
+
+        checkNotificationsPermissions()
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+
+        ViewTreeLifecycleOwner.get(this)?.lifecycle?.removeObserver(viewModel)
+
+        viewModel.removeNotifyMeListener()
+
+        listener = null
+
+        coroutineScope?.cancel()
+        coroutineScope = null
+    }
+
+    fun setListener(listener: NotifyMeListener?) {
+        this.listener = listener
+    }
+
+    fun setTitle(title: String) {
+        binding.notifyMeMessageTitle.text = title
+    }
+
+    fun setSubtitle(subtitle: String) {
+        binding.notifyMeMessageSubtitle.text = subtitle
+    }
+
+    private fun render(viewState: ViewState) {
+        if (viewState.visible) {
+            showMe()
+        } else {
+            hideMe()
+        }
+    }
+
+    private fun processCommands(command: Command) {
+        when (command) {
+            is CheckPermissions -> checkNotificationsPermissions()
+            is OpenSettings -> openSettings()
+            is Close -> hideMe()
+            is CheckShouldShowRequestPermissionRationale -> checkRequestPermissionRationale()
+            is ShowPermissionRationale -> showNotificationsPermissionsPrompt()
+        }
+    }
+
+    private fun checkNotificationsPermissions() {
+        if (isAtLeastAndroid13()) {
+            val value =
+                ContextCompat.checkSelfPermission(context, Manifest.permission.POST_NOTIFICATIONS) == PackageManager.PERMISSION_GRANTED
+            viewModel.updateNotificationsPermissions(value)
+        } else {
+            viewModel.updateNotificationsPermissions(true)
+        }
+    }
+
+    private fun openSettings() {
+        // This gets called only on Android 13+.
+        @SuppressLint("InlinedApi")
+        val settingsIntent = Intent(Settings.ACTION_APP_NOTIFICATION_SETTINGS)
+            .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            .putExtra(Settings.EXTRA_APP_PACKAGE, context.packageName)
+
+        startActivity(context, settingsIntent, null)
+    }
+
+    private fun showMe() {
```

Comment:
nit: Move to a similar interface as [`SearchBar`](https://github.com/duckduckgo/Android/blob/develop/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/SearchBar.kt#L27), where the Component handles actions and sends events.